cmd/query/app/querysvc/adjuster/spaniduniquifier_test.go

```diff
@@ -0,0 +1,108 @@
+)
+
+func makeTraces() ptrace.Traces {
+	traceID := pcommon.TraceID([]byte{0, 0, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 3})
+
+	traces := ptrace.NewTraces()
+	spans := traces.ResourceSpans().AppendEmpty().ScopeSpans().AppendEmpty().Spans()
+
+	clientSpan := spans.AppendEmpty()
+	clientSpan.SetTraceID(traceID)
+	clientSpan.SetSpanID(clientSpanID)
+	clientSpan.SetKind(ptrace.SpanKindClient)
+
+	serverSpan := spans.AppendEmpty()
+	serverSpan.SetTraceID(traceID)
+	serverSpan.SetSpanID(clientSpanID) // shared span ID
+	serverSpan.SetKind(ptrace.SpanKindServer)
+
+	anotherServerSpan := spans.AppendEmpty()
+	anotherServerSpan.SetTraceID(traceID)
+	anotherServerSpan.SetSpanID(clientSpanID) // shared span ID
+	anotherServerSpan.SetKind(ptrace.SpanKindServer)
+
+	anotherSpan := spans.AppendEmpty()
+	anotherSpan.SetTraceID(traceID)
+	anotherSpan.SetSpanID(anotherSpanID)
+
+	return traces
+}
+
+func TestSpanIDUniquifierTriggered(t *testing.T) {
+	traces := makeTraces()
+	deduper := SpanIDUniquifier()
+	require.NoError(t, deduper.Adjust(traces))
+
+	spans := traces.ResourceSpans().At(0).ScopeSpans().At(0).Spans()
+
+	clientSpan := spans.At(0)
+	assert.Equal(t, clientSpanID, clientSpan.SpanID(), "client span ID should not change")
+
+	serverSpan := spans.At(1)
+	assert.EqualValues(t, []byte{0, 0, 0, 0, 0, 0, 0, 2}, serverSpan.SpanID(), "server span ID should be reassigned")
+	assert.EqualValues(t, []byte{0, 0, 0, 0, 0, 0, 0, 3}, serverSpan.ParentSpanID(), "next server span should be this server span's parent")
+
+	anotherServerSpan := spans.At(2)
+	assert.EqualValues(t, []byte{0, 0, 0, 0, 0, 0, 0, 3}, anotherServerSpan.SpanID(), "server span ID should be reassigned")
+	assert.EqualValues(t, clientSpanID, anotherServerSpan.ParentSpanID(), "client span should be server span's parent")
+
+	thirdSpan := spans.At(3)
+	assert.Equal(t, anotherSpanID, thirdSpan.SpanID(), "3rd span ID should not change")
+}
+
+func TestSpanIDUniquifierNotTriggered(t *testing.T) {
+	traces := makeTraces()
+	spans := traces.ResourceSpans().At(0).ScopeSpans().At(0).Spans()
+
+	// only copy server span and random span
+	newSpans := ptrace.NewSpanSlice()
+	spans.At(1).CopyTo(newSpans.AppendEmpty())
+	spans.At(3).CopyTo(newSpans.AppendEmpty())
+	newSpans.CopyTo(spans)
+
+	deduper := SpanIDUniquifier()
+	require.NoError(t, deduper.Adjust(traces))
+
+	gotSpans := traces.ResourceSpans().At(0).ScopeSpans().At(0).Spans()
+
+	serverSpanID := clientSpanID // for better readability
+	serverSpan := gotSpans.At(0)
+	assert.Equal(t, serverSpanID, serverSpan.SpanID(), "server span ID should be unchanged")
+
+	thirdSpan := gotSpans.At(1)
+	assert.Equal(t, anotherSpanID, thirdSpan.SpanID(), "3rd span ID should not change")
+}
+
+func TestSpanIDUniquifierError(t *testing.T) {
+	traces := makeTraces()
+
+	maxID := pcommon.SpanID([8]byte{255, 255, 255, 255, 255, 255, 255, 255})
+
+	deduper := &spanIDDeduper{spansByID: make(map[pcommon.SpanID][]ptrace.Span), maxUsedID: maxID}
```

Comment:
```suggestion
	deduper := &spanIDDeduper{
		spansByID: make(map[pcommon.SpanID][]ptrace.Span),
		// instead of 0 start at the last possible value to cause an error
		maxUsedID: maxID,
	}
```